ambry-router/src/test/java/com.github.ambry.router/GetBlobOperationTest.java

```diff
@@ -474,6 +478,123 @@ public void testLegacyBlobGetSuccess()
   }
 
   /**
+   * Test range requests on a single chunk blob.
+   * @throws Exception
+   */
+  @Test
+  public void testRangeRequestSimpleBlob()
+      throws Exception {
+    // Random valid ranges
+    for (int i = 0; i < 5; i++) {
+      blobSize = random.nextInt(maxChunkSize) + 1;
+      int randomOne = random.nextInt(blobSize);
+      int randomTwo = random.nextInt(blobSize);
+      testRangeRequestClosedRange(Math.min(randomOne, randomTwo), Math.max(randomOne, randomTwo), true);
+    }
+
+    blobSize = random.nextInt(maxChunkSize) + 1;
+    // Entire blob
+    testRangeRequestClosedRange(0, blobSize - 1, true);
+    // Range that extends to end of blob
+    testRangeRequestOpenRange(random.nextInt(blobSize), true);
+    // Last n bytes of the blob
+    testRangeRequestLastNBytes(random.nextInt(blobSize) + 1, true);
+    // Last blobSize + 1 bytes (should not succeed)
+    testRangeRequestLastNBytes(blobSize + 1, false);
+    // Range over the end of the blob (should not succeed)
+    testRangeRequestClosedRange(random.nextInt(blobSize), blobSize + 5, false);
+    // Ranges that start past the end of the blob (should not succeed)
+    testRangeRequestOpenRange(blobSize, false);
+    testRangeRequestClosedRange(blobSize, blobSize + 20, false);
+    // 0 byte range
+    testRangeRequestLastNBytes(0, true);
```

Comment:
this is the only way to generate a 0 byte range, but I will add tests for various 1 byte ranges